src/main/java/com/aws/greengrass/builtin/services/telemetry/ComponentMetricIPCEventStreamAgent.java

```diff
@@ -0,0 +1,178 @@
+
+    @Getter(AccessLevel.PACKAGE)
+    private final Map<String, MetricFactory> metricFactoryMap = new HashMap<>();
+
+    @Inject
+    @Setter(AccessLevel.PACKAGE)
+    private AuthorizationHandler authorizationHandler;
+
+    @Inject
+    ComponentMetricIPCEventStreamAgent(AuthorizationHandler authorizationHandler) {
+        this.authorizationHandler = authorizationHandler;
+    }
+
+    public PutComponentMetricOperationHandler getPutComponentMetricHandler(
+            OperationContinuationHandlerContext context) {
+        return new PutComponentMetricOperationHandler(context);
+    }
+
+    class PutComponentMetricOperationHandler extends GeneratedAbstractPutComponentMetricOperationHandler {
+        private final String serviceName;
+
+        protected PutComponentMetricOperationHandler(OperationContinuationHandlerContext context) {
+            super(context);
+            serviceName = context.getAuthenticationData().getIdentityLabel();
+        }
+
+        @Override
+        protected void onStreamClosed() {
+            // NA
+        }
+
+        @SuppressWarnings({"PMD.PreserveStackTrace", "PMD.AvoidCatchingGenericException"})
+        @Override
+        public PutComponentMetricResponse handleRequest(PutComponentMetricRequest componentMetricRequest) {
+            return translateExceptions(() -> {
+                logger.atDebug().log("Received putComponentMetricRequest from component " + serviceName);
+
+                // Authorize request for given operation
+                try {
+                    doAuthorization(this.getOperationModelContext().getOperationName(), serviceName);
+                } catch (AuthorizationException e) {
+                    throw new UnauthorizedError(e.getMessage());
+                }
+
+                //validate - metric name length, value is non negative etc etc
+                List<software.amazon.awssdk.aws.greengrass.model.Metric> metricList =
+                        componentMetricRequest.getMetrics();
+                try {
+                    validateComponentMetricRequest(metricList);
+                } catch (IllegalArgumentException e) {
+                    throw new InvalidArgumentsError(e.getMessage());
+                }
+
+                // Perform translations on metrics List
+                try {
+                    final String metricNamespace = serviceName;
+                    translateAndEmit(metricList, metricNamespace);
+                } catch (IllegalArgumentException e) {
+                    throw new InvalidArgumentsError(e.getMessage());
+                } catch (Exception ex) {
+                    throw new ServiceError(ex.getMessage());
+                }
+
+                return new PutComponentMetricResponse();
+            });
+        }
+
+        @Override
+        public void handleStreamEvent(EventStreamJsonMessage streamRequestEvent) {
+            // NA
+        }
+    }
+
+    // Translate request metrics to telemetry metrics and emit them
```

Comment:
Add a java doc here instead of the a normal comment. As well as all the other functions.

Reply:
I can add java doc for public methods but we might not need them for all the private and internal methods which aren't consumed externally. What do you think?

Reply:
It just keeps the entire class much more cleaner to follow one style of doc.